0011-local-integration-test-framework.md

```diff
@@ -0,0 +1,103 @@
+
+## Detailed Design
+
+### Functional Requirements
+
+To implement a local backend for Lucy, we will use Docker containers to run the network configuration on a local machine. Docker containers are a lightweight way to run applications in an isolated environment. They are a great way to run applications in a consistent environment and are a great way to run applications locally. By using Docker containers, we can run the network configuration on a local machine without having to worry about the specifics of running the network on a local machine.
+
+We want the following properties for the local backend:
+
+- **Lightweight**: The local backend should be lightweight and should not require a lot of resources to run. This is important as we want to be able to run the network on a local machine without having to worry about the resource requirements of the network.
+- **Easy to use**: The local backend should be easy to use and should not require a lot of setup to run. This is important as we want to be able to run the network on a local machine without having to worry about the network setup.
+- **Fast**: The local backend should be fast and should not take a long time to run. This is important as we want to be able to run the network on a local machine without having to worry about the time it takes to run the network.
+
+### Orchestration
+
+To run containers locally, we will use [Docker Swarm](https://docs.docker.com/engine/swarm/) as the main driver. Docker Swarm is a container orchestration tool that allows us to run containers on a local machine or on a cluster of machines. It's built into the Docker Engine and there's no need for a seperate installation using it. The main reason to use this over [Docker Compose](https://docs.docker.com/compose/), is that Docker Swarm has multi-host networking built in, so if we want to further extend Lucy to run on a cluster of machines with Docker, we can do so easily. Docker Swarm takes a Docker Swarm file as input, which specifies all the containers to run and their configurations. To create a correct Docker Swarm file for our integration tests, we can create a mapping of the network configuration used to specify the tests and a Docker Swarm file. The network configuration is defined at the beginning of each integration test, which is then used to create a corresponding docker swarm file that has the specified network topology to use. We can use this file to specify the network topology configuration and then run the network on a local machine.
+
+Using Docker Swarm allows us to handle the orchestration of containers without having to worry about the specifics of running containers on a local machine. Docker Swarm will handle all the networking, log collection, and resource management for us.
+
+### Extending Lucy
```

Comment:
Sure thing! I added a note about the config type [here](https://github.com/o1-labs/rfcs/pull/40/files#diff-35f2db0a10b68cb10245a08c1cf9e572443feffe9fdd0f5032ade193bb604139R49)

The section for the pipe is already included, it will be shared with using volumes across each [service configuration](https://github.com/o1-labs/rfcs/pull/40/files#diff-35f2db0a10b68cb10245a08c1cf9e572443feffe9fdd0f5032ade193bb604139R109)